src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -170,22 +193,77 @@ class TaskDeploymentService {
-     * [taskRolePolicyString] with the provided [additionalPermissions]
+     * Helper method to initialise the lateinit vars [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] by
+     * converting the JSON to an instance of `JsonObject` data class and adding any requirements before serialising back to JSON
      *
-     * @return [Pair] of [taskRolePolicyString] to [taskAssumeRoleString] for ease of access.
+     * @return [taskRolePolicyString] and [jupyterBucketAccessRolePolicyString] respectively.
      */
-    fun parsePolicyDocuments(additionalPermissions: List<String>): Pair<String, String> {
-        var replaceString = ""
-        if (additionalPermissions.isNotEmpty()) {
-            logger.info("Adding permissions to containers", "permissions" to additionalPermissions.joinToString())
-            val permissionsJson = additionalPermissions.joinToString(prefix = "\"", separator = "\",\"", postfix = "\"")
-            replaceString = "$permissionsJson,"
+    fun parsePolicyDocument(resource: Resource, sidAndAdditions: Map<String, List<String>>, key: String): String{
+        val mapper = ObjectMapper()
+                .setPropertyNamingStrategy(CustomPropertyNamingStrategy())
+        val obj = mapper.readValue(resource.url, JsonObject::class.java)
+        obj.Statement.forEach {statement ->
+            sidAndAdditions.forEach {
+                if(it.key == statement.Sid) {
+                    if (key == "Resource") statement.Resource = statement.Resource.plus(it.value)
+                    else if (key == "Action") statement.Action = statement.Action.plus(it.value)
+                    else throw IllegalArgumentException("Key does not match expected values: \"Resource\" or \"Action\"")
+                }
+            }
+        }
+        return mapper.writeValueAsString(obj)
+    }
+
+/**
+ * Helper method to parse user access details into ARN format - returns list of these and the JupyterBucket ARN
+ */
+    fun createArnStringsList(pathPrefix: List<String>, pathSuffix: String, jupyterBucketArn: String): List<String>{
+        var returnList = listOf(jupyterBucketArn)
+        pathPrefix.forEach{
+            returnList = returnList.plus("arn:aws:kms:${configurationResolver.awsRegion}:${awsCommunicator.getAccNumber()}:alias/${it}-${pathSuffix}")
         }
-        taskAssumeRoleString = taskAssumeRoleDocument.inputStream.bufferedReader().use { it.readText() }
-        taskRolePolicyString = taskRolePolicyDocument.inputStream.bufferedReader().use { it.readText() }
-                .replace("ADDITIONAL_PERMISSIONS", replaceString)
-        return taskRolePolicyString to taskAssumeRoleString
+        return returnList
+    }
+}
+
+/**
+ * Class to override Jackson methods and ensure Proper Case is respected in serialisation of JSON keys - as required by AWS
+ */
+class CustomPropertyNamingStrategy : PropertyNamingStrategy() {
```

Comment:
This class should have a more descriptive name, `AwsPropertyNamingStrategy` maybe?